### ISSUE TITLE
Fedora 33 Build Error

### ISSUE THREAD

**Issue by ric96** [CURTLab/LVGLBuilder#8]
```
[ric@localhost LVGLBuilder]$ qmake-qt5 --version
QMake version 3.1
Using Qt version 5.15.1 in /usr/lib64
[ric@localhost LVGLBuilder]$ make
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DFT_CONFIG_OPTION_ERROR_STRINGS -DFT2_BUILD_LIBRARY -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I. -Ilvgl -Ifreetype/include -Ifreetype/builds/unix -I/usr/include/qt5 -I/usr/include/qt5/QtWidgets -I/usr/include/qt5/QtGui -I/usr/include/qt5/QtCore -Ibuild/moc -Ibuild/ui -I/../lib64/qt5/mkspecs/linux-g++ -o build/obj/LVGLPropertyRange.o properties/LVGLPropertyRange.cpp
In file included from properties/LVGLPropertyRange.cpp:1:
properties/LVGLPropertyRange.h:21:2: error: ‘LVGLPropertyMin’ does not name a type
   21 |  LVGLPropertyMin *m_min;
      |  ^~~~~~~~~~~~~~~
properties/LVGLPropertyRange.h:22:2: error: ‘LVGLPropertyMax’ does not name a type
   22 |  LVGLPropertyMax *m_max;
      |  ^~~~~~~~~~~~~~~
properties/LVGLPropertyRange.cpp: In constructor ‘LVGLPropertyRange::LVGLPropertyRange(LVGLProperty*)’:
properties/LVGLPropertyRange.cpp:41:4: error: class ‘LVGLPropertyRange’ does not have any field named ‘m_min’
   41 |  , m_min(new LVGLPropertyMin(this))
      |    ^~~~~
properties/LVGLPropertyRange.cpp:42:4: error: class ‘LVGLPropertyRange’ does not have any field named ‘m_max’
   42 |  , m_max(new LVGLPropertyMax(this))
      |    ^~~~~
properties/LVGLPropertyRange.cpp:44:14: error: ‘m_min’ was not declared in this scope
   44 |  m_childs << m_min << m_max;
      |              ^~~~~
properties/LVGLPropertyRange.cpp:44:23: error: ‘m_max’ was not declared in this scope
   44 |  m_childs << m_min << m_max;
      |                       ^~~~~
make: *** [Makefile:7263: build/obj/LVGLPropertyRange.o] Error 1
[ric@localhost LVGLBuilder]$ 
```

**Comment by CURTLab** [comment]
Should be fixed with https://github.com/CURTLab/LVGLBuilder/commit/ba8fc24fff0f76cd73304e804e1b78919c03ee7f

Please can you check it with your gcc version?

**Comment by ric96** [comment]
ba8fc24 resolved it but i have newer ones

```
[sahaj@athena LVGLBuilder]$ make
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DFT_CONFIG_OPTION_ERROR_STRINGS -DFT2_BUILD_LIBRARY -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I. -Ilvgl -Ifreetype/include -Ifreetype/builds/unix -I/usr/include/qt5 -I/usr/include/qt5/QtWidgets -I/usr/include/qt5/QtGui -I/usr/include/qt5/QtCore -Ibuild/moc -Ibuild/ui -I/../lib64/qt5/mkspecs/linux-g++ -o build/obj/LVGLPropertyVal2.o properties/LVGLPropertyVal2.cpp
In file included from properties/LVGLPropertyVal2.cpp:1:
properties/LVGLPropertyVal2.cpp: In instantiation of ‘LVGLPropertyVal2<T>::LVGLPropertyVal2(T, T, QString, std::function<T(_lv_obj_t*)>, T, T, QString, std::function<T(_lv_obj_t*)>, QString, std::function<void(_lv_obj_t*, T, T)>, QString, LVGLProperty*) [with T = short int]’:
properties/LVGLPropertyVal2.cpp:102:123:   required from here
properties/LVGLPropertyVal2.h:12:17: warning: unused parameter ‘functionName’ [-Wunused-parameter]
   12 |         QString functionName, std::function<void(lv_obj_t*, T, T)> setter,
      |         ~~~~~~~~^~~~~~~~~~~~
properties/LVGLPropertyVal2.cpp: In instantiation of ‘LVGLPropertyVal2<T>::LVGLPropertyVal2(T, T, QString, std::function<T(_lv_obj_t*)>, T, T, QString, std::function<T(_lv_obj_t*)>, QString, std::function<void(_lv_obj_t*, T, T)>, QString, LVGLProperty*) [with T = short unsigned int]’:
properties/LVGLPropertyVal2.cpp:109:124:   required from here
properties/LVGLPropertyVal2.h:12:17: warning: unused parameter ‘functionName’ [-Wunused-parameter]
properties/LVGLPropertyVal2.cpp: In instantiation of ‘void LVGLPropertyVal2<T>::LVGLPropertyVal::set(LVGLObject*, T) [with T = short unsigned int]’:
properties/LVGLPropertyVal2.cpp:49:6:   required from here
properties/LVGLPropertyVal2.cpp:51:93: error: invalid use of non-static data member ‘LVGLProperty::m_parent’
   51 |  LVGLPropertyVal2<T> *d = reinterpret_cast<LVGLPropertyVal2<T> *>(const_cast<LVGLProperty*>(m_parent));
      |                                                                                             ^~~~~~~~
In file included from properties/LVGLPropertyVal2.h:4,
                 from properties/LVGLPropertyVal2.cpp:1:
./LVGLProperty.h:42:22: note: declared here
   42 |  const LVGLProperty *m_parent;
      |                      ^~~~~~~~
properties/LVGLPropertyVal2.cpp: In instantiation of ‘void LVGLPropertyVal2<T>::LVGLPropertyVal::set(LVGLObject*, T) [with T = short int]’:
properties/LVGLPropertyVal2.cpp:49:6:   required from here
properties/LVGLPropertyVal2.cpp:51:93: error: invalid use of non-static data member ‘LVGLProperty::m_parent’
   51 |  LVGLPropertyVal2<T> *d = reinterpret_cast<LVGLPropertyVal2<T> *>(const_cast<LVGLProperty*>(m_parent));
      |                                                                                             ^~~~~~~~
In file included from properties/LVGLPropertyVal2.h:4,
                 from properties/LVGLPropertyVal2.cpp:1:
./LVGLProperty.h:42:22: note: declared here
   42 |  const LVGLProperty *m_parent;
      |                      ^~~~~~~~
make: *** [Makefile:7971: build/obj/LVGLPropertyVal2.o] Error 1
[sahaj@athena LVGLBuilder]$ 
```

Also GCC:
```
[sahaj@athena LVGLBuilder]$ gcc -v
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/10/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,d,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.1 20201016 (Red Hat 10.2.1-6) (GCC) 
[sahaj@athena LVGLBuilder]$ 
```

**Comment by CURTLab** [comment]
Please check the second fix: https://github.com/CURTLab/LVGLBuilder/commit/90ce3340db1a81c97cd1b1451f5b9b0e0bae3a71

**Comment by ric96** [comment]
> Please check the second fix: [90ce334](https://github.com/CURTLab/LVGLBuilder/commit/90ce3340db1a81c97cd1b1451f5b9b0e0bae3a71)

nope, similar error still now references updates lines.

```
[sahaj@athena LVGLBuilder]$ make
g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DFT_CONFIG_OPTION_ERROR_STRINGS -DFT2_BUILD_LIBRARY -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I. -Ilvgl -Ifreetype/include -Ifreetype/builds/unix -I/usr/include/qt5 -I/usr/include/qt5/QtWidgets -I/usr/include/qt5/QtGui -I/usr/include/qt5/QtCore -Ibuild/moc -Ibuild/ui -I/../lib64/qt5/mkspecs/linux-g++ -o build/obj/LVGLPropertyVal2.o properties/LVGLPropertyVal2.cpp
In file included from properties/LVGLPropertyVal2.cpp:1:
properties/LVGLPropertyVal2.cpp: In instantiation of ‘LVGLPropertyVal2<T>::LVGLPropertyVal2(T, T, QString, std::function<T(_lv_obj_t*)>, T, T, QString, std::function<T(_lv_obj_t*)>, QString, std::function<void(_lv_obj_t*, T, T)>, QString, LVGLProperty*) [with T = short int]’:
properties/LVGLPropertyVal2.cpp:102:123:   required from here
properties/LVGLPropertyVal2.h:12:17: warning: unused parameter ‘functionName’ [-Wunused-parameter]
   12 |         QString functionName, std::function<void(lv_obj_t*, T, T)> setter,
      |         ~~~~~~~~^~~~~~~~~~~~
properties/LVGLPropertyVal2.cpp: In instantiation of ‘LVGLPropertyVal2<T>::LVGLPropertyVal2(T, T, QString, std::function<T(_lv_obj_t*)>, T, T, QString, std::function<T(_lv_obj_t*)>, QString, std::function<void(_lv_obj_t*, T, T)>, QString, LVGLProperty*) [with T = short unsigned int]’:
properties/LVGLPropertyVal2.cpp:109:124:   required from here
properties/LVGLPropertyVal2.h:12:17: warning: unused parameter ‘functionName’ [-Wunused-parameter]
properties/LVGLPropertyVal2.cpp: In instantiation of ‘void LVGLPropertyVal2<T>::LVGLPropertyVal::set(LVGLObject*, T) [with T = short unsigned int]’:
properties/LVGLPropertyVal2.cpp:49:6:   required from here
properties/LVGLPropertyVal2.cpp:51:67: error: invalid use of non-static data member ‘LVGLProperty::m_parent’
   51 |  LVGLPropertyVal2<T> *d = reinterpret_cast<LVGLPropertyVal2<T> *>(m_parent);
      |                                                                   ^~~~~~~~
In file included from properties/LVGLPropertyVal2.h:4,
                 from properties/LVGLPropertyVal2.cpp:1:
./LVGLProperty.h:42:16: note: declared here
   42 |  LVGLProperty *m_parent;
      |                ^~~~~~~~
properties/LVGLPropertyVal2.cpp: In instantiation of ‘void LVGLPropertyVal2<T>::LVGLPropertyVal::set(LVGLObject*, T) [with T = short int]’:
properties/LVGLPropertyVal2.cpp:49:6:   required from here
properties/LVGLPropertyVal2.cpp:51:67: error: invalid use of non-static data member ‘LVGLProperty::m_parent’
   51 |  LVGLPropertyVal2<T> *d = reinterpret_cast<LVGLPropertyVal2<T> *>(m_parent);
      |                                                                   ^~~~~~~~
In file included from properties/LVGLPropertyVal2.h:4,
                 from properties/LVGLPropertyVal2.cpp:1:
./LVGLProperty.h:42:16: note: declared here
   42 |  LVGLProperty *m_parent;
      |                ^~~~~~~~
make: *** [Makefile:7971: build/obj/LVGLPropertyVal2.o] Error 1
[sahaj@athena LVGLBuilder]$ 
```

**Comment by CURTLab** [comment]
Please check if it works with the new fix
